npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/vulnerability_auditor.rb

```diff
@@ -15,9 +15,10 @@ module Dependabot
   module NpmAndYarn
     class UpdateChecker < Dependabot::UpdateCheckers::Base
       class VulnerabilityAuditor
-        def initialize(dependency_files:, credentials:)
+        def initialize(dependency_files:, credentials:, allow_removal: false)
```

Comment:
How will `allow_removal` be toggled?

Reply:
It's triggered based on the `npm_transitive_dependency_removal` option being present:
https://github.com/dependabot/dependabot-core/pull/5549/files#diff-90eca29ed297dba81a7ce1bc74ff4d3293e109bfa4e4183f4170d7322a9cbf27R115